src/v2/Apps/Settings/Routes/DeleteAccount/DeleteAccountRoute.tsx

```diff
@@ -1,83 +1,116 @@
+import { FC } from "react"
 import { RouterLink } from "v2/System/Router/RouterLink"
-import { deleteMyAccount } from "../../Mutations/DeleteMyAccount"
-import { useSystemContext } from "v2/System/SystemContext"
+import { Formik, Form } from "formik"
+import { useDeleteAccount } from "./useDeleteAccount"
+import { logout } from "v2/Utils/auth"
 
-export const DeleteAccountRoute = () => {
-  const { relayEnvironment } = useSystemContext()
-
-  const [isConfirmed, setConfirmed] = useState(false)
-  const [enteredReason, setEnteredReason] = useState("")
-  const isDisabled = !isConfirmed || !enteredReason
+export const DeleteAccountRoute: FC = () => {
   const { sendToast } = useToasts()
-
-  const handleDeleteMyAccount = async () => {
-    const url = window.location.href
-    try{
-      await deleteMyAccount(relayEnvironment!, enteredReason, url)
-      window.location.href = "/"
-      sendToast({
-        variant: "success",
-        message: "Profile deleted successfully",
-      })
-    } catch (error) {
-      const parsedError = JSON.parse(error.message)
-      sendToast({
-        variant: "error",
-        message: "There was a problem",
-        description: parsedError.error
-        ,
-      })
-    }
-  }
+  const { submitMutation } = useDeleteAccount()
 
   return (
-    <>
-        <Text 
-          color="black100" 
-          variant="lg"
-          mb={4}
-        >
+    <GridColumns>
+      <Column span={8}>
+        <Text variant="lg" mb={4}>
           Delete My Account
         </Text>
-        <Checkbox 
-          onSelect={(result) => {
-            setConfirmed(result)
-          }}  
-          selected={isConfirmed}
-          mb={4}
-        >
-          I understand that this will permanently delete my account and cannot be undone.
-        </Checkbox>
-        <TextArea
-          width="50%"
-          name="required-text-area"
-          title="Please Tell Us Why"
-          onChange={(result) => {
-            setEnteredReason(result.value)
+
+        <Formik
+          validateOnMount
+          initialValues={{
+            explanation: "",
+            confirmation: false,
+          }}
+          validationSchema={Yup.object().shape({
+            explanation: Yup.string().min(1).required(),
+            confirmation: Yup.boolean().oneOf([true]),
+          })}
+          onSubmit={async ({ explanation }) => {
+            try {
+              await submitMutation(
+                { input: { explanation, url: window.location.href } },
+                {
+                  checkForErrors: res => {
+                    return res.deleteMyAccountMutation?.userAccountOrError
+                      ?.mutationError
+                  },
+                }
+              )
+
+              sendToast({
+                variant: "success",
+                message: "Profile deleted successfully",
+                description: "Redirecting you to the Artsy homepage",
+              })
+
+              await logout()
```

Comment:
Previously this would redirect to the homepage and the recent bugfix to check for valid logins would then log you out and refresh. This avoids the extraneous refresh.